ppafm/io.py

```diff
@@ -471,224 +471,304 @@ def saveXSF(fname, data, lvec=None, dd=None, head=XSF_HEAD_DEFAULT, verbose=1 ):
+    F = readNumsUpTo(fname,nDim.astype(np.int32).copy(),noline)
+    if verbose: print("io | np.shape(F): ",np.shape(F))
+    if verbose: print("io | nDim: ", nDim)
+
+    FF = np.reshape(F, nDim)
+    if not xyz_order:
+        FF = FF.transpose((2,1,0)).copy()  # Transposition of the array to have the same order of data as in XSF file
+
+    nDim=[nDim[2],nDim[1],nDim[0]]                          # Setting up the corresponding dimensions.
+    head = []
+    head.append("BEGIN_BLOCK_DATAGRID_3D \n")
+    head.append("g98_3D_unknown \n")
+    head.append("DATAGRID_3D_g98Cube \n")
+    FF*=Hartree2eV
+    return FF,lvec, nDim, head
 
 # ================ WSxM output
 
 def saveWSxM_2D(name_file, data, Xs, Ys):
-	tmp_data=data.flatten()
-	out_data=np.zeros((len(tmp_data),3))
-	out_data[:,0]=Xs.flatten()
-	out_data[:,1]=Ys.flatten()
-	out_data[:,2]=tmp_data	#.copy()
-	f=open(name_file,'w')
-	print("WSxM file copyright Nanotec Electronica", file=f)
-	print("WSxM ASCII XYZ file", file=f)
-	print("X[A]  Y[A]  df[Hz]", file=f)
-	print("", file=f)
-	np.savetxt(f, out_data)
-	f.close()
+    tmp_data=data.flatten()
+    out_data=np.zeros((len(tmp_data),3))
+    out_data[:,0]=Xs.flatten()
+    out_data[:,1]=Ys.flatten()
+    out_data[:,2]=tmp_data    #.copy()
+    f=open(name_file,'w')
+    print("WSxM file copyright Nanotec Electronica", file=f)
+    print("WSxM ASCII XYZ file", file=f)
+    print("X[A]  Y[A]  df[Hz]", file=f)
+    print("", file=f)
+    np.savetxt(f, out_data)
+    f.close()
 
 def saveWSxM_3D( prefix, data, extent, slices=None ):
-	nDim=np.shape(data)
-	if slices is None:
-		slices=list(range( nDim[0]))
-	xs=np.linspace( extent[0], extent[1], nDim[2] )
-	ys=np.linspace( extent[2], extent[3], nDim[1] )
-	Xs, Ys = np.meshgrid(xs,ys)
-	for i in slices:
-		print("slice no: ", i)
-		fname = prefix+'_%03d.xyz' %i
-		saveWSxM_2D(fname, data[i], Xs, Ys)
+    nDim=np.shape(data)
+    if slices is None:
+        slices=list(range( nDim[0]))
+    xs=np.linspace( extent[0], extent[1], nDim[2] )
+    ys=np.linspace( extent[2], extent[3], nDim[1] )
+    Xs, Ys = np.meshgrid(xs,ys)
+    for i in slices:
+        print("slice no: ", i)
+        fname = prefix+'_%03d.xyz' %i
+        saveWSxM_2D(fname, data[i], Xs, Ys)
 
 # ================ Npy
 
-def saveNpy(fname, data, lvec , head=None):
-	np.save(fname+'.npy', data)
-	np.save(fname+'_vec.npy',lvec)
+def saveNpy(fname, data, lvec , atomic_info):
+    '''
+    Function for saving scalar grid data, together with its lattice_vector and information about original atoms and the original lattice vector (lvec0) in numpy format
+
+        Arguments:
+            fname: str. Path to file.
+            data: np.ndarray of shape (n_z, n_y, n_x) with scallar data
+            lvec: np.ndarray of shape (4, 3). Lattice vector of the data
+            atomic_info: tuple of shape (2). First part is [e, x, y, z] of atoms, the second is lvec of the atoms from the original geometry file, named as lvec0;
```

Comment:
The docstring system is unfortunately quite strict about the formatting of the docstring, so the `Arguments`  section should start at the same indentation level as the rest of lines so that it gets parsed correctly.

You can always test how it looks by building the docs locally: https://github.com/Probe-Particle/ppafm/issues/47#issuecomment-1327363260.